addons/udes_stock/tests/test_stock_move.py

```diff
@@ -493,6 +495,179 @@ def test_get_uncovered_moves_returns_move_if_only_partially_covered(self):
         self.assertEqual(moves.get_uncovered_moves(banana_5ml), moves)
 
 
+class TestUdesPropagateCancel(common.BaseUDES):
+    @classmethod
+    def setUpClass(cls):
+        """
+        Test u_propagate_cancel field is working on at least a 3 pick chain
+        """
+        super().setUpClass()
+        cls.Picking = cls.env["stock.picking"]
+        cls.Move = cls.env["stock.move"]
+        cls.package1 = cls.create_package()
+        cls.package2 = cls.create_package()
+        cls.create_trailer_route()
+        # Create a picking
+        cls._pick_info = [{"product": cls.banana, "uom_qty": 6}]
+        cls.quant1 = cls.create_quant(cls.banana.id, cls.test_stock_location_01.id, 5)
+        cls.quant2 = cls.create_quant(cls.banana.id, cls.test_stock_location_02.id, 3)
+        cls.pick = cls.create_picking(
+            cls.picking_type_pick, products_info=cls._pick_info, assign=True
+        )
+        cls.pick_ml1 = cls.pick.move_line_ids.filtered(lambda ml: ml.product_qty == 5)
+        cls.pick_ml2 = cls.pick.move_line_ids.filtered(lambda ml: ml.product_qty == 1)
+
+    @classmethod
+    def create_trailer_route(cls):
```

Comment:
I think this should be added to `UnconfiguredBaseUdes` as it seems quite general

Reply:
It's specific to this case, if i add it to there it will break most unit tests as they are testing without trailer at the end

Reply:
I think I came across something similar in a story. @PeterC10, maybe the routes in `UnconfiguredBaseUdes` need changing in the future , such that Trailer dispatch is included (so that the tests better reflect the actual routes in UDES)? should I create an issue for it?

Reply:
Yep make an issue and it can be discussed later.